mg.c

```diff
@@ -1240,7 +1240,13 @@ Perl_magic_get(pTHX_ SV *sv, MAGIC *mg)
                 Safefree(gary);
             }
         }
-        (void)SvIOK_on(sv);	/* what a wonderful hack! */
+
+        /*
+            Set this to avoid warnings when the SV is used as a number.
+            Avoid setting the public IOK flag so that serializers will
+            use the PV.
+        */
+        (void)SvIOKp_on(sv);	/* what a wonderful hack! */
```

Comment:
maybe keep IOK on if `num_groups == 1` ?

Reply:
I believe that would still be inconsistent with how Perl usually implements PVIV, which is that as soon as the PV mismatches the IV the IOK flag goes off.

In this case `1 1` doesn’t match 1. It does logically, but not at the level of the actual data value.

Reply:
That also wouldn’t fix the issue w/ serialization as it first appeared for us, which was that what JSON and Data::Dumper serialize as "$UID $UID" came out as $UID, which broke tests.